pallets/fusion/src/lib.rs

```diff
@@ -0,0 +1,2290 @@
+		NoActivePointsToUnbond,
+		/// The amount to unbond is invalid
+		InvalidUnbondAmount,
+		/// Unbonding this amount will make the remaining below minimum
+		AmountWillGoBelowMinimum,
+		/// User has too much unbonding chunks
+		MaxUnbondingChunksExceeded,
+		/// No funds are available to withdraw
+		NoFundsToWithdraw,
+		/// Caller is not authorized for this operation
+		NotAuthorized,
+		/// No rewards were found for the era
+		NoRewardsForEra,
+		/// The exposure is not founds
+		ExposureNotFound,
+		/// The user was not found in the exposure
+		UserNotFoundInExposure,
+		/// Rewards were already claimed for this user / era
+		AlreadyClaimed,
+		/// A user tried to unbond another user but it's only allowed if the pool is destroying
+		PoolIsNotDestroying,
+		/// Action is not allowed as the pool is destroying
+		PoolIsDestroying,
+		/// To handle compounding and easy bouding and unbonding, avail currency has no minimum
+		NoMinAmountForAvailCurrency,
+		/// There is no controller address to withdraw to
+		NoControllerAddressForUser,
+		/// If you're active points are below minimum, you cannot set compound to true
+		CannotSetCompoudingWithLessThanMinimum,
+		/// The state cannot be set to open if the pool is not nominating
+		PoolIsNotNominating,
+		/// The fusion currency rate was not found
+		FusionCurrencyRateNotFound,
+		/// The fusion pallet is paused so the operation is not allowed
+		FusionPalletPaused,
+		/// The controller of the slash destination can only be set with the correct extrinsic
+		CannotSetControllerForSlashDestination,
+		/// There are too many simultaneous slashes
+		TooManySlashes,
+		/// Invalid slash index
+		InvalidSlashIndex,
+		/// Invalid slash pool id
+		InvalidSlashPoolId,
+		/// TODO Temp
+		CannotDepositAvailCurrency,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// TODO - Dummy extrinsic to add currency without bridge, to be removed
+		#[pallet::call_index(99)]
+		#[pallet::weight(T::WeightInfo::create_currency())]
+		pub fn deposit_currency_dummy(
+			origin: OriginFor<T>,
+			evm_address: EvmAddress,
+			currency_id: CurrencyId,
+			amount: FusionCurrencyBalance,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			Self::do_deposit_currency(evm_address, currency_id, amount)?;
+			Ok(())
+		}
+
+		/// TODO - Dummy extrinsic to simulate an on_slash, to be removed
+		#[pallet::call_index(98)]
+		#[pallet::weight(T::WeightInfo::create_currency())]
+		pub fn dummy_slash(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+			bonded_amount: BalanceOf<T>,
+			slashed_amount: BalanceOf<T>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			Self::do_dummy_slash(who, bonded_amount, slashed_amount)?;
+			Ok(())
+		}
+
+		/// Pauses the entire pallet, halts rewards creation. Claiming, unbonding and withdrawing is still possible
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::create_currency())]
+		pub fn pause(origin: OriginFor<T>) -> DispatchResult {
+			ensure_root(origin)?;
```

Comment:
It is better to replace `ensure_root` in the pallet with `ApprovedOrigin` & configure it in the runtime